src/hooks/use-aggrid-row-filter.ts

```diff
@@ -119,11 +125,17 @@ export const useAggridRowFilter = (
 
             const { value, type, dataType } = selectedValue;
 
+            let backendValue = value;
+
+            if (dataType === 'number' && kiloValues.includes(field)) {
+                backendValue = String(parseInt(value) * 1000);
```

Comment:
The issue was not in the roundToPrecision() method.
you cannot use parseInt here : 2.5 => 2 => 2000